ethereum/eip712/eip712.go

```diff
@@ -16,475 +16,37 @@
 package eip712
 
 import (
-	"bytes"
-	"encoding/json"
-	"fmt"
-	"math/big"
-	"reflect"
-	"strings"
-	"time"
-
-	sdkmath "cosmossdk.io/math"
-	"github.com/cosmos/cosmos-sdk/crypto/keys/ed25519"
-	"golang.org/x/text/cases"
-	"golang.org/x/text/language"
-
-	errorsmod "cosmossdk.io/errors"
-	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
-	sdk "github.com/cosmos/cosmos-sdk/types"
-	errortypes "github.com/cosmos/cosmos-sdk/types/errors"
-
-	"github.com/ethereum/go-ethereum/common"
-	"github.com/ethereum/go-ethereum/common/math"
 	"github.com/ethereum/go-ethereum/signer/core/apitypes"
 )
 
-// WrapTxToTypedData is an ultimate method that wraps Amino-encoded Cosmos Tx JSON data
-// into an EIP712-compatible TypedData request.
+// WrapTxToTypedData wraps an Amino-encoded Cosmos Tx JSON SignDoc
+// bytestream into an EIP712-compatible TypedData request.
 func WrapTxToTypedData(
-	cdc codectypes.AnyUnpacker,
 	chainID uint64,
-	msg sdk.Msg,
 	data []byte,
-	feeDelegation *FeeDelegationOptions,
 ) (apitypes.TypedData, error) {
-	txData := make(map[string]interface{})
-
-	if err := json.Unmarshal(data, &txData); err != nil {
-		return apitypes.TypedData{}, errorsmod.Wrap(errortypes.ErrJSONUnmarshal, "failed to JSON unmarshal data")
-	}
-
-	domain := apitypes.TypedDataDomain{
-		Name:              "Cosmos Web3",
-		Version:           "1.0.0",
-		ChainId:           math.NewHexOrDecimal256(int64(chainID)),
-		VerifyingContract: "cosmos",
-		Salt:              "0",
+	messagePayload, err := createEIP712MessagePayload(data)
+	message := messagePayload.message
+	if err != nil {
+		return apitypes.TypedData{}, err
 	}
 
-	msgTypes, err := extractMsgTypes(cdc, "MsgValue", msg)
+	types, err := createEIP712Types(messagePayload)
 	if err != nil {
 		return apitypes.TypedData{}, err
 	}
 
-	if feeDelegation != nil {
```

Comment:
Remove `feePayer` since it's a legacy of the previous signature verification system.